src/security/Crypto.js

```diff
@@ -1,135 +1,183 @@
+            {
+                name: 'ECDSA',
+                namedCurve: 'P-256',
+            },
+            extractable,
+            ['sign', 'verify'],
+        );
+        keyPair.publicKey = new Uint8Array(await crypto.subtle.exportKey('spki', keyPair.publicKey));
+        if (extractable) {
+            keyPair.privateKey = new Uint8Array(await crypto.subtle.exportKey('pkcs8', keyPair.privateKey));
         }
-        const keyPair = nacl.sign.keyPair();
-        keyPair.id = base64Url(sha256(keyPair.publicKey)).substring(0, 16);
-        keyPair.algorithm = 'ED25519';
+        keyPair.id = base64Url(await crypto.subtle.digest('SHA-256', keyPair.publicKey)).substring(0, 16);
+        keyPair.algorithm = 'ECDSA_SHA256';
         keyPair.level = keyLevel;
         return keyPair;
     }
 
     /**
-     * Generates a temporary keypair to use with the token System
+     * Generates a temporary key pair to use with the Token system
      *
-     * @param {string} keyLevel - desired security level of key
+     * @param {string} keyLevel - "LOW", "STANDARD", or "PRIVILEGED"
      * @param {string} expirationMs - expiration date of the key in milliseconds
-     * @return {object} keyPair - keyPair
+     * @return {Object} generated key pair
      */
-    static generateTemporaryKeys(keyLevel, expirationMs) {
-        const keyPair = this.generateKeys(keyLevel);
+    static async generateTemporaryKeys(keyLevel, expirationMs) {
+        const keyPair = await this.generateKeys(keyLevel);
         keyPair.expiresAtMs = expirationMs;
         return keyPair;
     }
 
     /**
      * Signs a json object and returns the signature
      *
-     * @param {object} json - object to sign
-     * @param {object} keys - keys to sign with
-     * @return {string} signature - signature
+     * @param {Object} json - object to sign
+     * @param {Object} keys - keys to sign with
+     * @return {string} signature
      */
-    static signJson(json, keys) {
-        return Crypto.sign(stringify(json), keys);
+    static async signJson(json, keys) {
+        return await Crypto.sign(stringify(json), keys);
     }
 
     /**
-     * Signs a string and returns the signature
+     * Signs a string and returns the signature.
      *
      * @param {string} message - message to sign
-     * @param {object} keys - keys to sign with
-     * @return {string} signature - signature
+     * @param {Object} keys - keys to sign with
+     * @return {string} signature
      */
-    static sign(message, keys) {
+    static async sign(message, keys) {
+        if (!BROWSER) {
+            const msg = Crypto.wrapBuffer(message);
+            return base64Url(nacl.sign.detached(msg, keys.privateKey));
+        }
         const msg = Crypto.wrapBuffer(message);
-        return base64Url(nacl.sign.detached(msg, keys.secretKey));
+        let importedPrivateKey = keys.privateKey;
+        if (!importedPrivateKey instanceof CryptoKey) {
```

Comment:
when do we need imported keys that are not cryptokey?

Reply:
can only sign with CryptoKey object